week-3/todo-list/script.js

```diff
@@ -1,25 +1,108 @@
+
+    const span = document.createElement("span");
+    listItem.appendChild(span);
+    span.classList.add("badge", "bg-primary", "rounded-pill");
+
+    // Create and append a check icon
+    // const checkIcon = document.createElement("i");
+    // checkIcon.classList.add("fa", "fa-check");
+    // span.appendChild(checkIcon);
 
+    const crashIcon = document.createElement("i");
+    crashIcon.classList.add("fa", "fa-trash");
+    span.appendChild(crashIcon);
+
+    crashIcon.addEventListener("click", () => {
+      listItem.remove();
+      let index = todos.indexOf(item);
+      todos.splice(index, 1);
+    });
+
+    // Add a change event listener to the checkbox
+    checkBox.addEventListener("click", () => {
+      if (item.completed === false) {
+        listItem.classList.toggle("completed");
+        checkBox.classList.toggle("fa-solid");
+        checkBox.classList.toggle("fa-check");
+      }
+
+      item.completed = !item.completed;
+      // if (item.completed === false) {
+      //   item.completed = true;
+      // }
+    });
+  });
+}
 // These are the same todos that currently display in the HTML
 // You will want to remove the ones in the current HTML after you have created them using JavaScript
 let todos = [
-  { task: "Wash the dishes", completed: false },
-  { task: "Do the shopping", completed: false },
+  // { task: "Wash the dishes", completed: false },
+  // { task: "Do the shopping", completed: false },
 ];
-
 populateTodoList(todos);
-
 // This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
+const addButton = document.getElementById("add-button");
+addButton.addEventListener("click", addNewTodo);
+
 function addNewTodo(event) {
   // The code below prevents the page from refreshing when we click the 'Add Todo' button.
   event.preventDefault();
   // Write your code here... and remember to reset the input field to be blank after creating a todo!
+  const input = document.getElementById("input-todo");
+  if (input.value.length > 0) {
+    const newToDo = {
+      task: input.value,
+      completed: false,
+    };
+
+    todos.push(newToDo);
+    input.value = "";
+    populateTodoList(todos);
+  }
+
+  // console.log(todos);
 }
 
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
 function deleteAllCompletedTodos() {
   // Write your code here...
+  const deleteList = document.querySelectorAll(".completed");
+  deleteList.forEach((item) => {
+    // console.log(item);
+    // for (const todoElement of todos) {
+    //   if (item.textContent === todoElement.task) {
+    //     todoElement.completed = true;
+    //   }
+    // }
+    item.remove();
+  });
+
+  for (const element of todos) {
```

Comment:
If you were to use the filter function to remove completed todo items from `todos` array, how would you do that?